### PR TITLE
TextField and TextArea components adapted to the new design system

### PR DESCRIPTION
**Pull request by vmilan**
# Description

Shortcut: [#268673](https://app.shortcut.com/cartoteam/story/268673/components-basic-components-forms)
[sc-268673]

- TextField and TextArea components  with the new design system applied.
- Custom component to create Password Fields
- Also added some minor details to Tabs stories.

### CODE REVIEW

File: packages/react-ui/src/components/atoms/PasswordField.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      ref={ref}
+      style={style}
+      type={showPassword ? 'text' : 'password'}
+      InputProps={{
+        endAdornment: (
+          <InputAdornment position='end'>
+            <IconButton onClick={handleClickShowPassword}>
+              {showPassword ? <VisibilityOutlined /> : <VisibilityOffOutlined />}
+            </IconButton>
+          </InputAdornment>
```
Comment (VictorVelarde) [inline packages/react-ui/src/components/atoms/PasswordField.js:24]: Ok, I see the value here. At the beginning I thought we were just going to manage 'password' mode as hardcode. Obviously this makes more sense

File: packages/react-ui/src/components/atoms/PasswordField.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React, { forwardRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { IconButton, InputAdornment, TextField } from '@mui/material';
+import { VisibilityOffOutlined, VisibilityOutlined } from '@mui/icons-material';
+
+const PasswordField = forwardRef(({ style, ...otherProps }, ref) => {
```
Comment (VictorVelarde) [inline packages/react-ui/src/components/atoms/PasswordField.js:6]: Why do you extract here 'style' vs 'otherProps'

File: packages/react-ui/src/types.d.ts [hunk 3]
```diff
@@ -219,3 +219,8 @@ type TooltipDataProps = {
   ];
   title?: string;
 };
+
+// PasswordField
+export interface PasswordFieldProps {
```
Comment (VictorVelarde) [inline packages/react-ui/src/types.d.ts:224]: do we really need to declare this interface, couldn't we just allow same TextField props?

Comment (VictorVelarde) [reply packages/react-ui/src/types.d.ts:224]: See this custom TextField here: https://mui.com/material-ui/react-text-field/#customization (eg  RedditTextField). It basically passes down all props, using **TextFieldProps** .

What do you think?

File: packages/react-ui/src/components/atoms/PasswordField.js (hunk 2 above)
Comment (vmilan) [reply packages/react-ui/src/components/atoms/PasswordField.js:6]: you are right, is not needed. Removed!

File: packages/react-ui/src/types.d.ts (hunk 3 above)
Comment (vmilan) [reply packages/react-ui/src/types.d.ts:224]: yes, totally, removed.

Comment (vmilan) [reply packages/react-ui/src/types.d.ts]: I also cleaned up Button, which had the same unnecessary interface.

**Pull request merged**